File: Programmers_java/주식가격.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        Stack<Integer> stack = new Stack<>();
        for (int p : prices) {
            stack.add(p);
        }

        Deque<Integer> compare = new ArrayDeque<>();
        for (int i = stack.size() - 1; i >= 0; i--) {
            int cur = stack.pop();
            int cnt = 0;

            for (Integer c : compare) {
                cnt++;
                if (cur > c) break;
            }

            answer[i] = cnt;
            compare.addFirst(cur);
        }

        return answer;
    }
}

File: Programmers_js/주식가격.js
function solution(prices) {
    var answer = [];
    
    for(let i=0; i<prices.length-1; i++) {
        let time = 0;
        for(let j=i+1; j<prices.length; j++) {
            time += 1;
            if(prices[i] > prices[j]) {
                break;
            }
        }
        answer.push(time);
    }
    answer.push(0);
    
    return answer;
}
